http/parse.go

```diff
@@ -136,7 +144,8 @@ func Parse(r *http.Request, root *cmds.Command) (cmds.Request, error) {
 		return nil, err
 	}
 
-	return req, nil
+	err = req.FillDefaults()
+	return req, err
```

Comment:
So would we want to not return nil for the request if FillDefaults fails?

Reply:
No! We can't do that, because we want to print the help text if it was requested, even if some of the parsing failed. For that we need to return the incomplete request.